### ISSUE TITLE
Document the prototypes

### ISSUE THREAD

**Issue by zgrose** [vasl-developers/vasl#452]
Very hard to move/add/remove traits on counters without unintended side-effects. Would be nice to have a visual "inheritance tree" so we could see what's what at a glance.

**Comment by davidsullivan317** [comment]
Are you thinking something offline like CSS or something in the VASSAL piece editor?

**Comment by zgrose** [comment]
Not as a feature, just as a reference document. Once the skeleton is in place, maintainence should be pretty easy. Would help fixing the double HIP and extra HOB menu type issues.

**Comment by derimmer** [comment]
[VASL Prototypes.pdf](https://github.com/vasl-developers/vasl/files/1666286/VASL.Prototypes.pdf)

Zoltan

Is this the kind of thing you had in mind? A bit clunky perhaps and obviously incomplete.  Could be formatted differently but is the idea to show the prototype hierarchy? 

Doug

**Comment by bkemp01** [comment]
I stumbled on the attached files that I believe were made some time ago by Tom Repetti. Not sure if still current or how helpful they might be...
[VASL counter traits and prototypes.pptx](https://github.com/vasl-developers/vasl/files/1667735/VASL.counter.traits.and.prototypes.pptx)
[HR buildFile.txt](https://github.com/vasl-developers/vasl/files/1667738/HR.buildFile.txt)
[Prototypes.xlsx](https://github.com/vasl-developers/vasl/files/1667729/Prototypes.xlsx)

--
Brian

**Comment by zgrose** [comment]
That's the kind of thing (the PDF),yes . Didn't look at the others yet.

**Comment by derimmer** [comment]
Tom's powerpoint seems to be aimed at doing the same thing as the pdf, only formatted differently (and much more complete than my example). I have no preference as to format, happy to update Tom's document (if required) if that is the way to go. 

I think if I were a counter guy, it would be helpful to have a view that drilled down from the top counter level to the underlying prototypes. Tom's format seemed, to my eye, to go the other way. But, like I said, I am happy to use either format (or another if anyone else has ideas). 

Can anyone explain the link between the HR buildFile.txt and the actual buildFile?   Does the text file help us understand/decode the strings that appear in the buildFile?

The spreadsheet is probably a good source of new issues that would flow out of this documentation work. I have no idea if any of them have already been resolved.

**Comment by bkemp01** [comment]
I think HR=Human Readable. I'm guessing Tom wrote something to parse the buildFile into something understandable for spot checking errors. 

I don't have a preference on how this is approached, just passing along additional info in case it was useful.

**Comment by derimmer** [comment]
I am making some progress on this. Should have something for viewing in a couple of weeks.

**Comment by derimmer** [comment]
Is there a way to upload Tom's documents and my new draft into the Wik?. Dammed if I can figure out how. I can create a link but to where?

**Comment by derimmer** [comment]
Here is a complete draft of my prototype document, using all of the prototypes currently found in Game Piece Prototype Definitions folder visible when you choose Edit Module in VASSAL. If this seems both accurate and useful to people, I would suggest that the next step is to merge it with the Tom Repetti powerpoint document provided by Brian, . 

I think what my document does best is show the hierarchy within prototypes and perhaps highlights some differences, especially from one nationality to another, that may be worth reviewing. The document gets a little ragged at the end, where it shows prototypes that do not seem to be linked to other prototypes. There is nothing wrong with such a situation as I understand it; at some point prototypes are used by actual counters. 

[VASL Prototypes.pdf](https://github.com/vasl-developers/vasl/files/1818695/VASL.Prototypes.pdf)

**Comment by zgrose** [comment]
Beautiful. This will be super handy when fixing those "my [foo] is missing [bar]" bug reports.

**Comment by BigAl737** [comment]
This is fantastic Doug!

Sent from my iPad

**Comment by derimmer** [comment]
Glad to hear that you find it helpful. To that end, I have created a "Fun with Counters" page on the VASL Wiki and added links to my document and Tom's to that page. Using the Wiki should be more effective for discussing issues and how best to address them.

**Comment by bkemp01** [comment]
thanks for doing this Doug

unfortunately it will be obsolete when there is need for new prototypes, any way to automate updating the documentation?

FYI I added a new prototype to fix the Chinese Type 2 AC


________________________________

**Comment by derimmer** [comment]
agree with you Brian that one of the things we need to think about is how to keep the documentation current. Not sure at this point. How often do we add/change prototypes? Will add this to the Wiki page.

**Comment by bkemp01** [comment]
It seems pretty rare to add a prototype, but frequently fixes require fiddling with the hierarchy. Usually they just get added when a new nationality/core module comes along so this could be the case with Korea if we want to include those counters in VASL core.

**Comment by derimmer** [comment]
Both the document I produced and Tom Repetti's earlier one, entitled "Prototype issues and actions required" highlight a number of items that we might want to resolve. I have pulled all of those into the WikiPage, [Fun with Counters](https://github.com/vasl-developers/vasl/wiki/Fun-with-counters:-VASL-Prototypes), and started to annotate them and identify items we could create as issues.  

Additional comments, corrections and suggestions welcomed.
